bootloader: make pyi_arch_find_by_name case-insensitive on Windows

rokm opened the pull request:
Make `pyi_arch_find_by_name` archive helper case-insensitive on Windows for TOC entries that correspond to extractable files (e.g., BINARY, DATA, DEPENDENCY).

This fixes splash screen in onefile mode failing to find `VCRUNTIME140.dll` (as listed in the splash dependency list) in the archive, where it may be listed as `vcruntime140.dll`.

Fixes #8103.

rokm committed:
bootloader: make pyi_arch_find_by_name case-insensitive on Windows

Make `pyi_arch_find_by_name` archive helper case-insensitive on
Windows for TOC entries that correspond to extractable files
(e.g., BINARY, DATA, DEPENDENCY).

This fixes splash screen in onefile mode failing to find
`VCRUNTIME140.dll` (as listed in the splash dependency list) in
the archive, where it may be listed as `vcruntime140.dll`.

rokm committed:
bootloader: make MERGE entry extraction case-insensitive on Windows

The `_extract_dependency_from_archive` helper, used by MERGE
multipackage codepath, should use case-insensitive name comparison
on Windows, as to avoid potential mismatches due to file names not
being strictly case-normalized.

bwoodsend commented:
macOS also (usually) uses a case insensitive file system. Do we not need to worry about that too?

rokm commented:
> macOS also (usually) uses a case insensitive file system. Do we not need to worry about that too?

Now that you mention it, we probably do. (If only just for the sake of consistency and formalities).

rokm committed:
bootloader: extend case-insensitive archive search to macOS

Extend case-insensitive search for extractable archive entries to
macOS, as it has case-insensitive filesystem by default.
